build-scripts/rcm-guest/rhel7next.sh

```diff
@@ -0,0 +1,83 @@
+#!/bin/bash
+set -eu
+
+BASE_DIR=/mnt/redhat
+REMOTE_DIR=/srv/enterprise/rhel
+LATEST_RHEL_COMPOSE=${BASE_DIR}/nightly/latest-RHEL-7
+LATEST_EXTRAS=${BASE_DIR}/nightly/EXTRAS-RHEL-7.4/latest-EXTRAS-7-RHEL-7/compose/Server/x86_64/os
+REPO_FILE=rhel7next.repo
+MIRROR=use-mirror-upload.ops.rhcloud.com
+
+rsync=$(echo \
+    rsync \
+    --archive --hard-links --verbose --delete-after --progress --no-p --no-g \
+    --omit-dir-times --chmod=Dug=rwX)
+
+_reposync() {
+    local repoid
+    repoid=$1
+    reposync \
+        --config "${REPO_FILE}" \
+        --download_path "${repoid}/" --repoid "${repoid}" \
+        --arch x86_64 --download-metadata --downloadcomps --delete --norepopath
+}
+
+_createrepo_comp() {
+    local item
+    item=$1
+    createrepo \
+        --update --outputdir "${item}/" --group "${item}/comps.xml" "${item}/"
+}
+
+_createrepo() {
+    local item
+    item=$1
+    createrepo --update --outputdir "${item}/" "${item}/"
+}
+
+cat > rhel7next.repo <<-'EOF'
+	[rhel-7-fast-datapath-rpms]
+	name = Red Hat Enterprise Linux 7 Fast Datapath
+	baseurl = http://pulp.dist.prod.ext.phx2.redhat.com/content/dist/rhel/server/7/7Server/x86_64/fast-datapath/os/
+	enabled = 0
+	gpgcheck = 0
+
+	[rhel-7-fast-datapath-htb-rpms]
+	name = Red Hat Enterprise Linux 7 Fast HTB Datapath
+	baseurl = http://pulp.dist.prod.ext.phx2.redhat.com/content/htb/rhel/server/7/x86_64/fast-datapath/os/
+	enabled = 0
+	gpgcheck = 0
+
+	[rhel-7-server-ose-3.4-rpms]
+	name=rhel-7-server-ose-3.4-rpms
+	baseurl=http://pulp.dist.prod.ext.phx2.redhat.com/content/dist/rhel/server/7/7Server/x86_64/ose/3.4/os/
+	gpgcheck=0
+	enabled=0
+EOF
+# rsync Server to os mirror
+${rsync} \
+    "${LATEST_RHEL_COMPOSE}/compose/Server/x86_64/os/" \
+    "${MIRROR}:${REMOTE_DIR}/rhel7next/os/"
+# rsync Server-Optional to os mirror
+${rsync} \
+    "${LATEST_RHEL_COMPOSE}/compose/Server-optional/x86_64/os/" \
+    "${MIRROR}:${REMOTE_DIR}/rhel7next/optional/"
+# rsync Server-Extras to os mirror
+${rsync} \
+    "${LATEST_EXTRAS}/" \
+    "${MIRROR}:${REMOTE_DIR}/rhel7next/extras/"
+# reposync repos
+_reposync rhel-7-fast-datapath-rpms
+_reposync rhel-7-fast-datapath-htb-rpms
+_reposync rhel-7-server-ose-3.4-rpms
+# Rebuild repos with comp files
+_createrepo_comp "${PWD}/rhel-7-fast-datapath-rpms"
+_createrepo_comp "${PWD}/rhel-7-fast-datapath-htb-rpms"
+# Rebuild repos without comp files
+_createrepo "${PWD}/rhel-7-server-ose-3.4-rpms"
+# rsync repos to os mirror
+${rsync} \
+    rhel-7-{fast-datapath{,-htb},server-ose-3.4}-rpms \
+    "${MIRROR}:${REMOTE_DIR}/"
+# push to all geos
+ssh "${MIRROR}" /usr/local/bin/push.enterprise.sh -v
```

Comment:
Fought with a subtle error here before: when you stream a scriptA to ssh1 and then invoke ssh2 within scriptA, ssh2 will start eating stdin and read the rest of scriptA. If someone puts anything after this line in rhel7next.sh, it might not be executed because of this idiosyncrasy. Fix is here: https://github.com/openshift/aos-cd-jobs/blob/c3ae10a82e1fbbd74d1c384d926c8a799e9ede66/jobs/starter/package-overrides/scripts/update-cluster-overrides.sh#L75-L82  . Fixing it in invoke_rcm_guest by wrapping everything in { } < /dev/null should fix it for everything.

Reply:
oh my

Reply:
Makes sense.  Fixed (but it's not pretty).